src/vm/proftoeeinterfaceimpl.cpp

```diff
@@ -5082,36 +5081,46 @@ HRESULT ProfToEEInterfaceImpl::GetILToNativeMapping2(FunctionID functionId,
         return E_INVALIDARG;
     }
 
-    if (reJitId != 0)
-    {
-        return E_NOTIMPL;
-    }
-
-#ifdef DEBUGGING_SUPPORTED
-    // Cast to proper type
-    MethodDesc * pMD = FunctionIdToMethodDesc(functionId);
+    HRESULT hr = S_OK;
 
-    if (pMD->HasClassOrMethodInstantiation() && pMD->IsTypicalMethodDefinition())
+    EX_TRY
     {
-        // In this case, we used to replace pMD with its canonical instantiation
-        // (FindOrCreateTypicalSharedInstantiation).  However, a profiler should never be able
-        // to get to this point anyway, since any MethodDesc a profiler gets from us
-        // cannot be typical (i.e., cannot be a generic with types still left uninstantiated).
-        // We assert here just in case a test proves me wrong, but generally we will
-        // disallow this code path.
-        _ASSERTE(!"Profiler passed a typical method desc (a generic with types still left uninstantiated) to GetILToNativeMapping2");
-        return E_INVALIDARG;
-    }
+        // Cast to proper type
+        MethodDesc * pMD = FunctionIdToMethodDesc(functionId);
 
-    if (g_pDebugInterface == NULL)
-    {
-        return CORPROF_E_DEBUGGING_DISABLED;
+        if (pMD->HasClassOrMethodInstantiation() && pMD->IsTypicalMethodDefinition())
+        {
+            // In this case, we used to replace pMD with its canonical instantiation
+            // (FindOrCreateTypicalSharedInstantiation).  However, a profiler should never be able
+            // to get to this point anyway, since any MethodDesc a profiler gets from us
+            // cannot be typical (i.e., cannot be a generic with types still left uninstantiated).
+            // We assert here just in case a test proves me wrong, but generally we will
+            // disallow this code path.
+            _ASSERTE(!"Profiler passed a typical method desc (a generic with types still left uninstantiated) to GetILToNativeMapping2");
+            hr = E_INVALIDARG;
+        }
+        else
+        {
+            CodeVersionManager *pCodeVersionManager = pMD->GetCodeVersionManager();
+            ILCodeVersion ilCodeVersion = NULL;
+            {
+                CodeVersionManager::TableLockHolder lockHolder(pCodeVersionManager);
+
+                pCodeVersionManager->GetILCodeVersion(pMD, reJitId);
+            }
+        
```

Comment:
Lets add another comment to indicate this is an arbitrarily chosen version and profilers are recommended to use GetILToNativeMapping3 to avoid that ambiguity.

Reply:
Ok, sounds good